parquet2hive_modules/parquet2hivelib.py

```diff
@@ -204,62 +219,76 @@ def avro2sql(avro, name, version, location, partitions, with_version=True):
+    field_names = [field[2] for field in fields]
     duplicate_columns = set(field_names) & set(partitions)
     assert not duplicate_columns, "Columns {} are in both the table columns and the partitioning columns; they should only be in one or another".format(", ".join(duplicate_columns))
-    table_name = name + "_" + version if with_version else name
-    return "drop table if exists {0}; create external table {0}({1}){2} stored as parquet location '\"'{3}/{4}'\"'; msck repair table {0};".format(table_name, fields_decl, partition_decl, location, version)
-
-
-def avro2sql_column(avro):
-    return "`{}` {}".format(avro["name"], transform_type(avro["type"]))
-
-
-def transform_type(avro):
-    is_dict, is_list, is_str = isinstance(avro, dict), isinstance(avro, list), isinstance(avro, str) or isinstance(avro, unicode)
-
-    unchanged_types = ['string', 'int', 'float', 'double', 'boolean', 'date', 'timestamp', 'binary']
-    mapped_types = {'integer': 'int', 'long': 'bigint'}
-
-    if is_str and avro in unchanged_types:
-        sql_type = avro
-    elif is_str and avro in mapped_types:
-        sql_type = mapped_types[avro]
-    elif is_dict and avro["type"] == "map":
-        value_type = avro.get("values", avro.get("valueType"))  # this can differ depending on the Avro schema version
-        sql_type = "map<string,{}>".format(transform_type(value_type))
-    elif is_dict and avro["type"] == "array":
-        item_type = avro.get("items", avro.get("elementType"))  # this can differ depending on the Avro schema version
-        sql_type = "array<{}>".format(transform_type(item_type))
-    elif is_dict and avro["type"] in ("record", "struct"):
-        fields_decl = ", ".join(["`{}`: {}".format(field["name"], transform_type(field["type"])) for field in avro["fields"]])
-        sql_type = "struct<{}>".format(fields_decl)
-        if avro["type"] == "record":
-            udf[avro["name"]] = sql_type
-    elif is_list:
-        sql_type = transform_type(avro[0] if avro[1] == "null" else avro[1])
-    elif avro in udf:
-        sql_type = udf[avro]
-    else:
-        raise Exception("Unknown type {}".format(avro))
+    return "drop table if exists {0}; create external table {0}({1}){2} stored as parquet location '\"'{3}'\"'; msck repair table {0};".format(table_name, fields_decl, partition_decl, location)
+
+
+def build_tree(schema, children):
+    retval = []
+
+    for _ in range(children):
+        elem = schema.pop(0)
+
+        elem_type = 'group' if elem.type is None else Type._VALUES_TO_NAMES[elem.type].lower()
+        converted = None if elem.converted_type is None else ConvertedType._VALUES_TO_NAMES[elem.converted_type].lower()
+        repetition = FieldRepetitionType._VALUES_TO_NAMES[elem.repetition_type].lower()
+
+        if elem_type == 'group':
+            subs = build_tree(schema, elem.num_children)
+            retval.append((repetition, elem_type, elem.name, converted, subs))
```

Comment:
Why not have this instead append a dict, and label each element (e.g. {"repetition": repetition, "type": elem_type, etc.}). Would make this, and the rest of the code, more readable.